### PR TITLE
Fix volume name prefix

### LINKED ISSUE #99: Volume prefix should append a hyphen if missing to match the CCM
If I have a volume claim named `foo` and put set the prefix env var to `bar`, then my display name in OCI will be `barfoo`. It should be `bar-foo` which matches the CCM logic. 

https://github.com/oracle/oci-cloud-controller-manager/blob/3d533ef04eb8233ddc3a730c8281742b4d5ae03a/pkg/oci/load_balancer_util.go#L412

### PR DESCRIPTION
**Pull request by harveylowndes**
Fixes error in PR #107 for fix #99

### CODE REVIEW

**Comment by MadalinaPatrichi** [comment]
Need to make the same changes to the fss and @prydie was mentioning that there could be backwards compatibility issues

File: pkg/provisioner/block/block.go [hunk 1]
```diff
@@ -163,10 +163,15 @@ func (block *blockProvisioner) Provision(options controller.VolumeOptions, ad *i
 
 	glog.Infof("Creating volume size=%v AD=%s compartmentOCID=%q", volSizeMB, *ad.Name, block.client.CompartmentOCID())
 
+	prefix := strings.TrimSpace(os.Getenv(volumePrefixEnvVarName))
+	if prefix != "" && !strings.HasSuffix(prefix, "-") {
+		prefix = fmt.Sprintf("%s%s", prefix, "-")
```
Comment (MadalinaPatrichi) [inline pkg/provisioner/block/block.go]: I would rename this to prefixedName or sth similar

Comment (harveylowndes) [reply pkg/provisioner/block/block.go]: Since this is only the prefix and not the prefix + the name, I think prefix will suffice.

**Comment by harveylowndes** [comment]
@MadalinaPatrichi yeah I believe at the time this was decided to be ok and perhaps reason for the initial merge. This will not have conflicts where `volumePrefixEnvVarName` doesnt exist. Although I agree this should be investigated.

File: pkg/provisioner/block/block.go (hunk 1 above)
Comment (MadalinaPatrichi) [reply pkg/provisioner/block/block.go]: Ah, yes, you're right, sorry, hadn't read this correctly

**Comment by prydie** [comment]
@MadalinaPatrichi Having given it some more thought I don't think there's a backwards compatibility problem as we don't ever look up by display name IIRC? If you agree then LGTM - please merge away 👍

**Pull request merged**